src/projects/shared/api/circuit/tests/public/Selections.test.ts

```diff
@@ -4,40 +4,329 @@ import "../Extensions";
+            const filtered = selections.filter((obj) => obj.name === "filterMe");
+            expect(filtered).toHaveLength(1);
+            expect(filtered[0]).toBe(c1);
+        });
+        test(".every()", () => {
+            const [{selections, undo, redo}, { }, {PlaceAt, GetPort}] = CreateTestRootCircuit();
+            const [c1, c2] = PlaceAt(V(0, 0), V(1, 1));
+            c1.select();
+            c1.name = "filterMe";
+            c2.select();
+            expect(selections.every((obj) => obj.name === "filterMe")).toBeFalsy();
+            expect(selections.every((obj) => obj.kind === "TestComp")).toBeTruthy();
+        });
     });
 
+    describe("Observe", () => {
+        test("Basic selecting and undo/redo", () => {
+            const [{selections, undo, redo}, { }, {PlaceAt, GetPort}] = CreateTestRootCircuit();
+            const [c1, c2] = PlaceAt(V(0, 0), V(1, 1));
+            let observedCount = 0;
+            selections.observe(() => {observedCount++});
+            c1.select();
+            expect(observedCount).toBe(1);
+            c2.select();
+            expect(observedCount).toBe(2);
+            undo();
+            undo();
+            redo();
+            redo();
+            expect(observedCount).toBe(6);
+        })
+    })
+
     describe("Obj Queries", () => {
-        // Test ideas:
-        // .all
-        // .components filters correctly
-        // .wires filters correctly
+        const [{selections}, { }, {PlaceAt, Connect, GetPort}] = CreateTestRootCircuit();
+        const [c1, c2, c3] = PlaceAt(V(0, 0), V(1, 1), V(2, 2));
+        const w1 = Connect(c1, c2);
+        const w2 = Connect(c1, c3);
+        const p1 = GetPort(c1);
+        const p2 = GetPort(c2);
+        const p3 = GetPort(c3);
+        c1.select();
+        c2.select();
+        p3.select();
+        w1.select();
+        test(".all", () => {
+            expect(selections.all).toHaveLength(4);
+            expect(selections.all).toContain(c1);
+            expect(selections.all).toContain(c2);
+            expect(selections.all).toContain(p3);
+            expect(selections.all).toContain(w1);
+        });
+        test(".components", () => {
+            expect(selections.components).toHaveLength(2);
+            expect(selections.components).toContain(c1);
+            expect(selections.components).toContain(c2);
+        });
+        test(".wires", () => {
+            expect(selections.wires).toHaveLength(1);
+            expect(selections.wires).toContain(w1);
+        });
     });
 
     describe("Midpoint", () => {
-        // Test ideas:
-        // Will have to do some math (don't have to do anything crazy)
+        test("2 components selected", () => {
+            const [{selections}, { }, {PlaceAt, Connect, GetPort}] = CreateTestRootCircuit();
+            const [c1, c2] = PlaceAt(V(0, 0), V(2, 2));
+            c1.select();
+            c2.select();
+            expect(selections.midpoint()).toStrictEqual(V(1, 1));
```

Comment:
I think just `.toEqual` is fine? But actually when it comes to math like this, we should use `toApproximatelyEqual`